doc/rst/language/spec/data-parallelism.rst

```diff
@@ -782,25 +782,21 @@ which results in the promoted expression:
 
    [b in B] A[b]
 
-However, it is an error to modify promoted expressions like this one.
-For example, the following is an error:
+Modifying promoted expressions may introduce undesirable race conditions in
+code. For example, the following code will potentially result in an incorrect
+result:
 
 .. code-block:: chapel
 
+   B = [1, 2, 1];
    A[B] += 3;
 
-If this was promoted, it would become the following:
+To avoid this race, the above code could be written using an explicit loop
+statement and the proper intents, for example:
 
 .. code-block:: chapel
 
-   [b in B] A[b] += 3;
-
-This is illegal, as ``A`` cannot be modified without an explicit ``ref`` intent.
-An explicit loop statement must be used, for example:
-
-.. code-block:: chapel
-
-   [b in B with (ref A)] A[b] += 3;
+   [b in B with (+ reduce A)] A[b] += 3;
```

Comment:
I'm not sure the reduce intent works for an entire array like this. Does this line compile?

Reply:
Yes it does. The following compiles and runs as expected.

```chapel
var A = [ii in 1..10] ii;
var B = [1,1,2];

[b in B with (+ reduce A)] A[b] += 3;
writeln(A);
```

It doesn't work when the bracket loop is expression level due to parsing issues, but a statement level bracket loop like this works the same as forall.

I can lock this in with a test if you prefer

Reply:
I don't think that's necessary, just wanted to make sure it would compile. Thanks!